payjoin/src/send/v2/error.rs

```diff
@@ -60,3 +60,59 @@ impl From<ParseReceiverPubkeyParamError> for CreateRequestError {
         CreateRequestError(InternalCreateRequestError::ParseReceiverPubkey(value))
     }
 }
+
+/// Error returned for v2-specific payload encapsulation errors.
+#[derive(Debug)]
+pub struct EncapsulationError {
+    internal: InternalEncapsulationError,
+}
```

Comment:
In other places where we wrap an internal type we just use a tuple struct:

```suggestion
pub struct EncapsulationError(InternalEncapsulationError)
```